src/js/components/Button/Button.js

```diff
@@ -202,8 +203,9 @@ const Button = forwardRef(
 
     const onMouseOutButton = event => {
       setHover(false);
-      if (onMouseOut) {
-        onMouseOut(event);
+      const outEvent = onMouseLeave || onMouseOut;
```

Comment:
What are your thoughts on handling `onMouseLeave` separately from `onMouseOut`?

Reply:
So @ShimiSun, `onMouseLeave` is used by `Tip` but button expects to receive an `onMouseOut` and I decided to stay with both because I was worried about backwards compatible, I saw that both are used in other files. If I remove `onMouseOut` some components that send it to `Button` could break.  

What do you think?

Reply:
I think it makes sense.  Thank you for the detailed answer.

Reply:
Given that the events are actually different, based on https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseleave_event, it doesn't seem correct to be merging the two here. I think having both wired up would allow the caller to get either behavior they want correctly.

Reply:
I agree @ericsoderberghp that this is not the best solution, but I couldn't figure out how to fix this without merging them. If I left only `onMouseOut` the problem will continue, btw the MDN docs that you sent it's a great resource to understand why the problem happens in the first place. 

If I left just `onMouseLeave` this could break some apps that are sending a custom `onMouseOut` function or some components on grommet that use this prop, like `AccordionPanel` .

I'm open for better ideas on how to handle this, thanks

Reply:
From Button's point of view, it should pass along either or both of those listeners and caller's should be responsible to listen as needed. Is that not viable?

Reply:
I noticed in the Codesandbox demo on the original issue (https://codesandbox.io/s/grommet-v2-template-forked-wlr22?file=/index.js) that the tooltip on the button with custom label only appears if I hover directly over the label but not if I just hover somewhere else on the disabled button. This makes me think the issue is rooted in how Button states are passed along to custom elements, like a custom label.

Is it possible instead to make sure that if button is `disabled` that no mouse events are registered on any of the button's children/custom elements? It seems that as long as the mouseover is not registered on the custom label, then we wouldn't be seeing this issue. If we could do that, we wouldn't need to merge the `onMouseOut` and `onMouseLeave` events.

Reply:
@ericsoderberghp Do you mean pass both props `onMouseOver` and `onMouseOut` to button? I tried that and didn't work out, has the same bug behaviour of the demo.

Reply:
@taysea To do this I think we will have to use something like `cloneElement` to pass information to a custom element in label or something else, I don't know if this is viable and we also have to know if it is a react element or no wich we don't need RN. See code bellow on how is implemented today.
https://github.com/grommet/grommet/blob/b68eac70c8e2c6a6b0013ac9f4c606dca6c6164a/src/js/components/Button/Button.js#L227-L244

Reply:
I was thinking Button would support either `onMouseOver` or `onMouseOut` or both and that Tip would use whatever was appropriate.

Reply:
Thanks for the reply @netochaves -- maybe exploring Eric's proposal would be more viable given the current implementation.

Reply:
@ericsoderberghp I said `onMouseOver` but I meant `onMouseLeave` if I pass that event and the `onMouseOut` the tip behaviour continues like the issue demo.